usr/share/rear/restore/COVE/default/400_restore_with_cove.sh

```diff
@@ -0,0 +1,333 @@
+#
+# Restore files and folders with Cove
+#
+
+# ANSI color escape sequences
+readonly RED='\033[0;31m'
+readonly GREEN='\033[0;32m'
+readonly NC='\033[0m' # No color
+
+readonly COVE_CLIENT_TOOL="${COVE_INSTALL_DIR}/bin/ClientTool"
+readonly COVE_TMPDIR="${TARGET_FS_ROOT}/covetmp-$(date +"%y%m%d%H%M%S")"
+
+COVE_INSTALLER_PATH="${COVE_TMPDIR}/mxb-linux-x86_64.run"
+COVE_INSTALLATION_TOKEN=""
+
+# Waits until a condition succeeds
+# $1: condition command
+# $2: interval between checks
+function cove_wait_for() {
+    local condition="$1"
+    local interval="$2"
+    while true; do
+        if eval "${condition}"; then
+            break
+        fi
+        sleep "${interval}"
+    done
+}
+
+# Prints a message without appending a newline
+# $@: message to print
+function cove_print() {
+    { printf "$*" 1>&7 || true ; } 2>>"/dev/$DISPENSABLE_OUTPUT_DEV"
+}
+
+# Prints "Done!" message
+function cove_print_done() {
+    cove_print "${GREEN}Done!${NC}\n"
+}
+
+# Prints "Error!" message
+function cove_print_error() {
+    cove_print "${RED}Error!${NC}\n"
+}
+
+# Prompts the user with yes/no question and handles default responses
+# $1: Prompt message to display
+# $2: Default value (y or n) if the user just presses Enter
+function cove_ask() {
+    local message="$1"
+    local default_value="$2"
+    local value=""
+    while true; do
+        read -r -p "${message} (y/n) [${default_value}]: " value 0<&6 1>&7 2>&8
+        value="${value:-$default_value}"
+        case "$value" in
+            [yY][eE][sS]|[yY])
+                return 0
+                ;;
+            [nN][oO]|[nN])
+                return 1
+                ;;
+            *)
+                 UserOutput "Invalid input. Please answer y/n."
+                ;;
+        esac
+    done
+}
+
+# Gets the BackupFP status
+function cove_get_status() {
+    "${COVE_CLIENT_TOOL}" control.status.get
+}
+
+# Downloads the Backup Manager installer
+function cove_download_bm_installer() {
+    if [ -z "${COVE_INSTALLER_URL}" ]; then
+        UserOutput ""
+        UserOutput "Please provide the URL to download the Backup Manager installer:"
+        read -r -p "URL: " COVE_INSTALLER_URL 0<&6 1>&7 2>&8
```

Comment:
Normally we use the UserInput function for such things,
see usr/share/rear/lib/_framework-setup-and-functions.sh and
https://github.com/rear/rear/wiki/Coding-Style#what-to-do-with-stdin-stdout-and-stderr